src/main/java/com/twilio/oai/TwilioPhpGenerator.java

```diff
@@ -1,40 +1,88 @@
+import com.twilio.oai.template.PHPAPIActionTemplate;
+import io.swagger.v3.oas.models.OpenAPI;
+import org.openapitools.codegen.CodegenModel;
+import org.openapitools.codegen.CodegenOperation;
 import org.openapitools.codegen.languages.PhpClientCodegen;
+import org.openapitools.codegen.model.ModelMap;
 import org.openapitools.codegen.model.ModelsMap;
+import org.openapitools.codegen.model.OperationsMap;
 
-import java.util.HashMap;
-import java.util.Map;
+import java.util.*;
 
 public class TwilioPhpGenerator extends PhpClientCodegen {
+    private String PHP_CONVENTIONAL_MAP_PATH = "config/" + EnumConstants.Generator.TWILIO_PHP.getValue() + ".json";
     private final TwilioCodegenAdapter twilioCodegen;
+    private final DirectoryStructureService directoryStructureService = new DirectoryStructureService(
+            additionalProperties,
+            new ResourceMap(new Inflector()),
+            new PhpCaseResolver());
+    private final List<CodegenModel> allModels = new ArrayList<>();
+    private final Map<String, String> modelFormatMap = new HashMap<>();
+    private IConventionMapper conventionMapper = new LanguageConventionResolver(PHP_CONVENTIONAL_MAP_PATH);
 
     public TwilioPhpGenerator() {
         super();
-
         twilioCodegen = new TwilioCodegenAdapter(this, getName());
     }
 
+    @Override
+    public String toApiFilename(final String name) {
+        return directoryStructureService.toApiFilename(super.toApiFilename(name));
+    }
+
     @Override
     public void processOpts() {
         super.processOpts();
-
-        apiTemplateFiles.clear();
-        apiTestTemplateFiles.clear();
-        modelTestTemplateFiles.clear();
-        apiDocTemplateFiles.clear();
-        modelDocTemplateFiles.clear();
         twilioCodegen.processOpts();
     }
 
     @Override
-    public Map<String, ModelsMap> postProcessAllModels(Map<String, ModelsMap> objs) {
+    public void processOpenAPI(final OpenAPI openAPI) {
+        final String domain = twilioCodegen.getDomainFromOpenAPI(openAPI);
+        twilioCodegen.setDomain(domain);
+        directoryStructureService.configure(openAPI);
+    }
+
+    @Override
+    public Map<String, ModelsMap> postProcessAllModels(Map<String, ModelsMap> allModels) {
+        final Map<String, ModelsMap> results = super.postProcessAllModels(allModels);
+
+        Utility.addModelsToLocalModelList(results, this.allModels);
+        Utility.setComplexDataMapping(this.allModels, this.modelFormatMap);
+
+        // Return an empty collection so no model files get generated.
         return new HashMap<>();
     }
 
+    @Override
+    public OperationsMap postProcessOperationsWithModels(final OperationsMap objs, List<ModelMap> allModels) {
+        final OperationsMap results = super.postProcessOperationsWithModels(objs, allModels);
+        final List<CodegenOperation> opList = directoryStructureService.processOperations(results);
+        APIResources apiResources = processCodegenOperations(opList);
```

Comment:
I like how clean this is. Should extend the refactoring to Node so we can see how this limits duplication as part of this review.